config-builder/src/main/java/com/ca/apim/gateway/cagatewayconfig/tasks/zip/builder/IdentityProviderEntityBuilder.java

```diff
@@ -35,25 +38,27 @@
     }
 
     private Entity buildIdentityProviderEntity(String name, IdentityProvider identityProvider) {
-        final Element identityProviderElement = document.createElement("l7:IdentityProvider");
+        final Element identityProviderElement = document.createElement(ID_PROV);
         final String id = idGenerator.generate();
-        identityProviderElement.setAttribute("id", id);
-        identityProviderElement.appendChild(DocumentTools.createElementWithTextContent(document, "l7:Name", name));
-        identityProviderElement.appendChild(DocumentTools.createElementWithTextContent(document, "l7:IdentityProviderType", identityProvider.getType().getValue()));
+        identityProviderElement.setAttribute(ATTRIBUTE_ID, id);
+        identityProviderElement.appendChild(DocumentTools.createElementWithTextContent(document, NAME, name));
+        identityProviderElement.appendChild(DocumentTools.createElementWithTextContent(document, ID_PROV_TYPE, identityProvider.getType().getValue()));
+        if (identityProvider.getProperties() != null) {
+            identityProviderElement.appendChild(BuilderUtils.buildPropertiesElement(identityProvider.getProperties(), document));
+        }
 
         switch(identityProvider.getType()) {
             case BIND_ONLY_LDAP:
-                if (identityProvider.getProperties() != null) {
-                    identityProviderElement.appendChild(BuilderUtils.buildPropertiesElement(
-                            identityProvider.getProperties().entrySet()
-                                    .stream()
-                                    .collect(Collectors.toMap(stringStringEntry -> "property." + stringStringEntry.getKey(), Map.Entry::getValue)), document));
-                }
                 identityProviderElement.appendChild(buildBindOnlyLdapIPDetails(identityProvider));
                 break;
             case LDAP:
             case INTERNAL:
             case FEDERATED:
+                final FederatedIdentityProviderDetail identityProviderDetail = (FederatedIdentityProviderDetail) identityProvider.getIdentityProviderDetail();
+                if (identityProviderDetail != null) {
```

Comment:
I think you should move this if statement to inside the method and rename it, so you leave the switch-case simpler and easier to read